### PR TITLE
lep: add local-volume

### PR DESCRIPTION
**Pull request by derekbit**
[longhorn #3957](longhorn/longhorn#3957)

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

File: enhancements/20221123-local-volume.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+## Motivation
+### Goals
+- Longhorn can support local volume (data locality=strict-local) for providing better IO latencies and IOPS.
+- A local volume can only have one replica.
+- A local volume supports the operations such as snapshot, backup and etc.
+
+### Non-goals
+- A local volume's data locality cannot be converted to other modes when volume is not detached.
+- A local volume does not support multiple replicas.
```
Comment (innobead) [inline enhancements/20221123-local-volume.md:17]: note: local replication could be an improvement in the future.

File: enhancements/20221123-local-volume.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+### Validating Webhook
+- If a volume with `strict-local` data locality, the `numberOfReplicas` should be 1.
+- If a local volume is attached, the conversion between `strict-local` and other data localities is not allowable.
+- If a local volume is attached, the update of the replica count is not allowable.
+
+### Test Plan
+#### Integration tests
+1. Successfully create a local volume with `numberOfReplicas=1` and `dataLocality=strict-local`.
+2. Check the validating webhook can reject the following cases
```
Comment (innobead) [inline enhancements/20221123-local-volume.md:65]:
```suggestion
2. Check the validating webhook can reject the following cases when the volume is created or attached
```

**Review by innobead — APPROVED** [review]
LGTM

File: enhancements/20221123-local-volume.md (hunk 2 above)
Comment (derekbit) [reply enhancements/20221123-local-volume.md]: Updated. Thank you.

File: enhancements/20221123-local-volume.md (hunk 1 above)
Comment (derekbit) [reply enhancements/20221123-local-volume.md]: Updated. Thank you.

**Review by innobead — APPROVED** [review]

**Pull request merged**